astropy/cosmology/parameter/_descriptors.py

```diff
@@ -0,0 +1,47 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import annotations
+
+__all__ = []
+
+from dataclasses import dataclass
+from types import MappingProxyType
+from typing import TYPE_CHECKING, Any, NoReturn
+
+from astropy.utils.compat.misc import PYTHON_LT_3_10
+
+if TYPE_CHECKING:
+    from astropy.cosmology.core import Cosmology
+
+
+_dataclass_kwargs = {} if PYTHON_LT_3_10 else {"slots": True}
+
+
+@dataclass(frozen=True, **_dataclass_kwargs)
+class ParameterDescriptor:
```

Comment:
But the easiest solution is just not to make this a `dataclass` - which would be logical since there are no data attributes! Instead, just define `__slots__ = ["attr_name"]`

Reply:
the `dataclass` gives the `slots` and also freezes the class, which is nice given how important the `parameters` property is. `ruff` will clean everything up when we're py3.10+. I switched `attr_name` to be a non-init'ed field.
Also, not that I expect anyone to use these, but `dataclass` also makes a better `repr` and `eq`.